jest/jestSetup.js

```diff
@@ -45,6 +45,13 @@ jest.mock('react-native', () => {
       resourceURL:
         'file:///private/var/containers/Bundle/Application/4DCD4D2B-F745-4C70-AD74-8E5F690CF593/ZulipMobile.app/',
     };
+  } else if (ReactNative.Platform.OS === 'android') {
+    const header = 'z|mock|';
+    ReactNative.NativeModules.TextCompressionModule = {
+      header,
+      compress: async (input: string) => `${header}${input}`,
+      decompress: async (input: string) => input.replace(header, ''),
```

Comment:
This is fine as is, but I'd be a bit more comfortable with something that mangles the whole string. Possibly base64encoding it, in addition to adding the header? (I'm not sure what JS runtime our jest tests execute in, so I'm not sure if we have a reasonable base64 function easily available to us, but if we do, that seems like a good option).

Reply:
Sounds good; there is such a thing, I think. I can use `Buffer.from`.

The Jest tests execute in Node, and we currently recommend the latest 12.x and use that in CI (see f83ccc6cc6ccfc3cb843428006f231b1d057346d for what this means more concretely). #M4263 is open to upgrade to 14.